# Determining the Best Location for a Start-Up Bar

Shannon Bee

December 22, 2020

## 1. Introduction

Start-up companies face several decisions, one of the biggest being location. The location of a business can make or break the company, and should be carefully determined in order to achieve the optimal conditions.  While all start-up businesses face challenges, this is especially in the restaurant/bar industry. Per an article by CNBC, around 60 percent of new restaurants fail within the first year, and nearly 80 percent shutter before their fifth anniversary. This analysis will specifically target the best location for a start-up bar, between San Francisco, CA and San Diego, CA.

#### 1.1 Background

The data that will be used in this analysis will be the number of similar bars in each city, the prices of drinks in each city, the costs of products needed for the bar based on the city, and the general population and demographic of each city.

#### 1.2 Problem

The objective of this analysis will be to determine if it will be more profitable to open a bar in San Diego, CA or San Francisco, CA. This will be determined by prices of similar products in the area, competitors, and the general population and demographic of the areas. The audience will be a start-up bar interested in maximizing profits and catering to a younger crowd. 

#### 1.3 Interest

The information will be presented to a start-up bar determining where to open their new venture.

## 2. Data Acquisition

This data will be obtained from Foursquare location data and other relevant datasets found online. For example, coordinates will be obtained from Foursquare location data to see surrounding competing bars and the popularity and prices for each. Based on this information, the most profitable location of the start-up bar can be determined, as well as estimated prices for products in order to be competitive.

#### 2.1 Data Sources

Foursquare location data

## 3. Exploratory Data Analysis

Population and age demographic information was obtained and detailed below

#### 3.1 Population and Age Demographics

San Diego has a population of 1,423,852 with a median age of 35.4. Per censusreporter.org, 23% is under 19, 18% is between 20 and 29, 17% is between 30 and 39, 13% is between 40 and 49, and 29% is over 50.

San Francisco has a population of 881,549 with a median age of 38.2. Per censusreporter.org, 15% is under 19, 17% is between 20 and 29, 21% is between 30 and 39, 13% is between 40 and 49, and 33% is over 50.

#### 3.2 Prices and Costs

Per thrillist.com, the average cost of a bulk bottle of alcohol in the state of California is 16.99 for Smirnoff and 30.00 for Grey Goose. Per expatisan.com, the average cocktail price in San Diego is 14. The average cocktail price in San Francisco is 15, indicating a slightly profitability in San Francisco. This will be analyzed and verified through Foursquare in the Predictive Modeling section below.

## 4. Predictive Modeling

#### 4.1 Surrounding Areas

In [33]:
import requests # library to handle requests
import pandas as pd # library for data analsysis
import numpy as np # library to handle data in a vectorized manner
import random # library for random number generation


!pip install geopy
from geopy.geocoders import Nominatim # module to convert an address into latitude and longitude values

# libraries for displaying images
from IPython.display import Image 
from IPython.core.display import HTML 
    
# tranforming json file into a pandas dataframe library
from pandas.io.json import json_normalize


! pip install folium==0.5.0
import folium # plotting library

print('Folium installed')
print('Libraries imported.')

Folium installed
Libraries imported.


In [34]:
CLIENT_ID = 'QHQN04UQO0ZGDXGQKDJWWQBP2OUJ2F4XGSDSVUGCQT5EH0E2' # your Foursquare ID
CLIENT_SECRET = 'C5ICXXSTRBLC1NDPEPOMJM4RA0ZMFD3DFLQPCXRBTPUUKWJB' # your Foursquare Secret
ACCESS_TOKEN = '02DRMR5KMEAV04C1PQJQXXS5GM4QTCHALZ51QXRXTYT4ISLG' # your FourSquare Access Token
VERSION = '20180604'
LIMIT = 30
print('Your credentails:')
print('QHQN04UQO0ZGDXGQKDJWWQBP2OUJ2F4XGSDSVUGCQT5EH0E2' + CLIENT_ID)
print('C5ICXXSTRBLC1NDPEPOMJM4RA0ZMFD3DFLQPCXRBTPUUKWJB' + CLIENT_SECRET)

Your credentails:
QHQN04UQO0ZGDXGQKDJWWQBP2OUJ2F4XGSDSVUGCQT5EH0E2QHQN04UQO0ZGDXGQKDJWWQBP2OUJ2F4XGSDSVUGCQT5EH0E2
C5ICXXSTRBLC1NDPEPOMJM4RA0ZMFD3DFLQPCXRBTPUUKWJBC5ICXXSTRBLC1NDPEPOMJM4RA0ZMFD3DFLQPCXRBTPUUKWJB


In [35]:
address = 'San Diego, CA'

geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

32.7174202 -117.1627728


In [36]:
address = 'San Francisco, CA'

geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

37.7790262 -122.4199061


In [80]:
search_query = 'Bar'
radius = 500
print(search_query + ' .... OK!')

Bar .... OK!


In [81]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&oauth_token={}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude,ACCESS_TOKEN, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=QHQN04UQO0ZGDXGQKDJWWQBP2OUJ2F4XGSDSVUGCQT5EH0E2&client_secret=C5ICXXSTRBLC1NDPEPOMJM4RA0ZMFD3DFLQPCXRBTPUUKWJB&ll=37.7790262,-122.4199061&oauth_token=02DRMR5KMEAV04C1PQJQXXS5GM4QTCHALZ51QXRXTYT4ISLG&v=20180604&query=Bar&radius=500&limit=30'

In [56]:
results = requests.get(url).json()
results

{'meta': {'code': 200, 'requestId': '5fe267c3dad389062d321c11'},
 'notifications': [{'type': 'notificationTray', 'item': {'unreadCount': 0}}],
 'response': {'venues': [{'id': '5905515f8c812a4130298084',
    'name': 'Thomas L Horn Bar & Lounge',
    'location': {'address': '401 Van Ness Ave',
     'lat': 37.77958,
     'lng': -122.42211,
     'labeledLatLngs': [{'label': 'display',
       'lat': 37.77958,
       'lng': -122.42211}],
     'distance': 203,
     'postalCode': '94102',
     'cc': 'US',
     'city': 'San Francisco',
     'state': 'CA',
     'country': 'United States',
     'formattedAddress': ['401 Van Ness Ave', 'San Francisco, CA 94102']},
    'categories': [{'id': '4bf58dd8d48988d123941735',
      'name': 'Wine Bar',
      'pluralName': 'Wine Bars',
      'shortName': 'Wine Bar',
      'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/winery_',
       'suffix': '.png'},
      'primary': True}],
    'referralId': 'v-1608673219',
    'hasPerk': False},
   {'id

In [82]:
# assign relevant part of JSON to venues
venues = results['response']['venues']

# tranform venues into a dataframe
dataframe = pd.json_normalize(venues)
dataframe.head()

id                        name  \
0  5905515f8c812a4130298084  Thomas L Horn Bar & Lounge   
1  5fbf51f35e4a23458f548454       Dickey's Barbecue Pit   

                                          categories    referralId  hasPerk  \
0  [{'id': '4bf58dd8d48988d123941735', 'name': 'W...  v-1608673219    False   
1  [{'id': '4bf58dd8d48988d1c4941735', 'name': 'R...  v-1608673219    False   

   location.address  location.lat  location.lng  \
0  401 Van Ness Ave     37.779580   -122.422110   
1  590 Van Ness Ave     37.780831   -122.419959   

                             location.labeledLatLngs  location.distance  ...  \
0  [{'label': 'display', 'lat': 37.77958, 'lng': ...                203  ...   
1  [{'label': 'display', 'lat': 37.7808311, 'lng'...                200  ...   

  location.state location.country  \
0             CA    United States   
1             CA    United States   

                     location.formattedAddress delivery.id  \
0  [401 Van Ness Ave, San Francisco, CA 94102]         NaN   
1  [590 Van Ness Ave, San Francisco, CA 94102]     2423749   

                                        delivery.url delivery.provider.name  \
0                                                NaN                    NaN   
1  https://www.grubhub.com/restaurant/dickeys-bar...                grubhub   

              delivery.provider.icon.prefix delivery.provider.icon.sizes  \
0                                       NaN                          NaN   
1  https://fastly.4sqi.net/img/general/cap/                     [40, 50]   

               delivery.provider.icon.name venuePage.id  
0                                      NaN          NaN  
1  /delivery_provider_grubhub_20180129.png    600684723  

[2 rows x 23 columns]

In [83]:
# keep only columns that include venue name, and anything that is associated with location
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
dataframe_filtered = dataframe.loc[:, filtered_columns]

# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

# filter the category for each row
dataframe_filtered['categories'] = dataframe_filtered.apply(get_category_type, axis=1)

# clean column names by keeping only last term
dataframe_filtered.columns = [column.split('.')[-1] for column in dataframe_filtered.columns]

dataframe_filtered

name  categories           address        lat  \
0  Thomas L Horn Bar & Lounge    Wine Bar  401 Van Ness Ave  37.779580   
1       Dickey's Barbecue Pit  Restaurant  590 Van Ness Ave  37.780831   

          lng                                     labeledLatLngs  distance  \
0 -122.422110  [{'label': 'display', 'lat': 37.77958, 'lng': ...       203   
1 -122.419959  [{'label': 'display', 'lat': 37.7808311, 'lng'...       200   

  postalCode  cc           city state        country  \
0      94102  US  San Francisco    CA  United States   
1      94102  US  San Francisco    CA  United States   

                              formattedAddress                        id  
0  [401 Van Ness Ave, San Francisco, CA 94102]  5905515f8c812a4130298084  
1  [590 Van Ness Ave, San Francisco, CA 94102]  5fbf51f35e4a23458f548454

## 5. Conclusions

San Francisco has around the same number of bars as San Diego. However, drinks are priced higher, resulting in a higher profit margin assuming the bulk CA costs. San Francisco also has a higher percentage of the population between the ages of 20 and 39 (38%) versus San Diego at 35%, which is the age range the most commonly frequents bars. 

## 6. Future Directions

Based on the analysis and conclusions, the new bar should be opened in San Francisco, CA to maximize profitability and target their age demographic. 